parachains/runtimes/assets/asset-hub-westend/src/xcm_config.rs

```diff
@@ -469,12 +482,19 @@ impl xcm_executor::Config for XcmConfig {
 	>;
 	type Trader = (
 		UsingComponents<WeightToFee, WestendLocation, AccountId, Balances, ToStakingPot<Runtime>>,
-		cumulus_primitives_utility::TakeFirstAssetTrader<
-			AccountId,
-			AssetFeeAsExistentialDepositMultiplierFeeCharger,
-			TrustBackedAssetsConvertedConcreteId,
-			Assets,
+		cumulus_primitives_utility::SwapFirstAssetTrader<
+			Runtime,
+			LocationToAccountId,
+			pallet_asset_conversion::Pallet<Runtime>,
+			WeightToFee,
+			LocalAndForeignAssetsConvertedConcreteId,
+			LocalAndForeignAssets<
```

Comment:
also `LocalAndForeignAssets` facade implementation for `Assets` / `ForeignAssets` is kind of questionable if it is 100% correct,
here I have two things:

1. everywhere there is a IF `if isLocal then Assets else ForeignAssets`, but if we use some `PoolAssets` asset, it will fall to `ForeignAssets` branch and yes happily it fails, but it is not 100% bullet proof this `if-else` way -> another vote for tuple implementation

2. it implements `Unbalanced` / `Balanced` traits, e.g.  I had a bug with that, because it was missing overridden implementation for `decrease_balance / increase_balance`, which caused bug for benchmarks:

```
bparity@bkontur-ThinkPad-P14s-Gen-2i:~/parity/cumulus/cumulus$ ARTIFACTS_DIR=tmp2 ../../command-bot-scripts/commands/bench/lib/bench-all-cumulus.sh 
[+] Compiling benchmarks...
[+] Listing pallets for runtime asset-hub-westend for chain: asset-hub-westend-dev ...
[+] Benchmarking 17 pallets for runtime asset-hub-westend for chain: asset-hub-westend-dev, pallets:
   [+] cumulus_pallet_xcmp_queue
   [+] frame_system
   [+] pallet_asset_conversion
   [+] pallet_assets
   [+] pallet_balances
   [+] pallet_collator_selection
   [+] pallet_multisig
   [+] pallet_nft_fractionalization
   [+] pallet_nfts
   [+] pallet_proxy
   [+] pallet_session
   [+] pallet_timestamp
   [+] pallet_uniques
   [+] pallet_utility
   [+] pallet_xcm
   [+] pallet_xcm_benchmarks::fungible
   [+] pallet_xcm_benchmarks::generic
2023-07-12 10:20:15 a defensive failure has been triggered; please report the block number at https://github.com/paritytech/substrate/issues: "write_balance is not used if other functions are impl'd"    
2023-07-12 10:20:15 panicked at 'Expected Ok(_). Got Err(
    Unavailable,
)', /home/bparity/.cargo/git/checkouts/substrate-7e08433d4c370a21/edf58dc/frame/asset-conversion/src/benchmarking.rs:60:9   
```


it took me a while to find a issue with this missing overridden implementations https://github.com/paritytech/cumulus/pull/2860/commits/b8f86cf03e136ddc78a896aafa8f4e56f0b57b12

and I am afraid that there could be possible other hidden bugs (maybe not),
so if we wanna go with this `LocalAndForeignAssets` we have to add there at least substrate **conformance tests** or whatever to ensure that `LocalAndForeignAssets` is 100% compatible with `Assets` and `ForeignAssets`